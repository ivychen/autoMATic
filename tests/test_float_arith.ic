/* autoMATic Compiler - Spring 2018 - COMS4115 PLT
 by Ivy Chen ic2389, Nguyen Chi Dung ncd2118,
 Nelson Gomez ng2573, Jimmy O'Donnell jo2474 */

int main() {
	float x;
	float y;
	float z;
	float w;

	x = 0.0;
	y = 1.0;
	if (y == 1.0) {
		println("y == 1");
	}

	x = x + y;
	println(x); // 1.0

	x = -x;
	println(x); // -1.0

	z = x / 2.0;
	println(z); // -0.5

	w = x * 5.0 - (2.0 / 2.0);
	println(w); // -6.0

	if (w < z) println("w < z");
	if (w > z) println("w > z");

	if (w >= z) println("w >= z");
	if (z >= w) println("z >= w");

	if (z <= w) println("z <= w");
	if (w <= z) println("w <= z");

	if (z <= z) println("z <= z");

	if (w == z) println("w == z");
	if (z == z) println("z == z");
	if (w == w) println("w == w");
	if (w != z) println("w != z");

        x = 2.0;
        println(x ** 2); // 4.0
        println(x ** 3.0); // 8.0
        println(x ** 4.0); // 16.0

        println(16.0 % 5.0); // 1.0
        println(2 ** 3.0); // 8.0

	return 0;
}
