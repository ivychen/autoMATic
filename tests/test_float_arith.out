y == 1
1.000000
-1.000000
-0.500000
-6.000000
w < z
z >= w
w <= z
z <= z
z == z
w == w
w != z
